etc/kayobe/ansible/scripts/smartmon.py

```diff
@@ -0,0 +1,156 @@
+        for attr in json_data['ata_smart_attributes']['table']:
+            attr_name = attr['name'].replace('-', '_').lower()
+            if attr_name in smartmon_attrs:
+                attr_id = attr.get('id', '')
+                value = attr.get('value', '')
+                worst = attr.get('worst', '')
+                threshold = attr.get('thresh', '')
+                raw_value = attr.get('raw', {}).get('value', '')
+                metrics.append(f"{attr_name}_value{{{labels},smart_id=\"{attr_id}\"}} {value}")
+                metrics.append(f"{attr_name}_worst{{{labels},smart_id=\"{attr_id}\"}} {worst}")
+                metrics.append(f"{attr_name}_threshold{{{labels},smart_id=\"{attr_id}\"}} {threshold}")
+                metrics.append(f"{attr_name}_raw_value{{{labels},smart_id=\"{attr_id}\"}} {raw_value}")
+    return metrics
+
+def parse_smartctl_info(disk, disk_type, json_data):
+    info = json_data.get('device', {})
+    smart_status = json_data.get('smart_status', {})
+    labels = {
+        'disk': disk,
+        'type': disk_type,
+        'vendor': info.get('vendor', ''),
+        'product': info.get('product', ''),
+        'revision': info.get('revision', ''),
+        'lun_id': info.get('lun_id', ''),
+        'model_family': json_data.get('model_family', ''),
+        'device_model': json_data.get('model_name', ''),
+        'serial_number': json_data.get('serial_number', '').lower(),
+        'firmware_version': json_data.get('firmware_version', '')
+    }
+    label_str = ','.join(f'{k}="{v}"' for k, v in labels.items())
+    metrics = [
+        f'device_info{{{label_str}}} 1',
+        f'device_smart_available{{disk="{disk}",type="{disk_type}",serial_number="{labels["serial_number"]}"}} {1 if smart_status.get("available", False) else 0}',
+    ]
+    if smart_status.get("available", False):
+        metrics.append(f'device_smart_enabled{{disk="{disk}",type="{disk_type}",serial_number="{labels["serial_number"]}"}} {1 if smart_status.get("enabled", False) else 0}')
+        if 'passed' in smart_status:
+            metrics.append(f'device_smart_healthy{{disk="{disk}",type="{disk_type}",serial_number="{labels["serial_number"]}"}} {1 if smart_status.get("passed", False) else 0}')
+    return metrics
+
+def format_output(metrics):
+    output = []
+    last_metric = ""
+    for metric in sorted(metrics):
+        metric_name = metric.split('{')[0]
+        if metric_name != last_metric:
+            output.append(f"# HELP smartmon_{metric_name} SMART metric {metric_name}")
+            output.append(f"# TYPE smartmon_{metric_name} gauge")
+            last_metric = metric_name
+        output.append(f"smartmon_{metric}")
+    return '\n'.join(output)
+
+def main():
+    try:
+        version_output = run_command([SMARTCTL_PATH, '-j'], parse_json=True)
+        smartctl_version_list = version_output.get('smartctl', {}).get('version', [])
+        if smartctl_version_list:
```

Comment:
Have you looked at Python libraries for doing this?

Eg. This one looks reasonable: https://pypi.org/project/pySMART/

It can handle loading all the device metrics in via `smartctl` as you are doing here. 

It seems to provide some abstraction over the health state as well.

The work you've put into the tests should be directly usable